include/exadg/fluid_structure_interaction/driver_precice_fluid.h

```diff
@@ -0,0 +1,594 @@
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    // writing
+    this->precice =
+      std::make_shared<ExaDG::preCICE::Adapter<dim, dim, VectorType>>(this->precice_parameters,
+                                                                      this->mpi_comm);
+
+    this->precice->add_write_surface(this->application->get_boundary_descriptor_fluid()
+                                       ->velocity->dirichlet_cached_bc.begin()
+                                       ->first,
+                                     this->precice_parameters.write_mesh_name,
+                                     {this->precice_parameters.stress_data_name},
+                                     this->precice_parameters.write_data_type,
+                                     fluid_matrix_free,
+                                     fluid_operator->get_dof_index_velocity(),
+                                     fluid_operator->get_quad_index_velocity_linear());
+
+    // structure to ALE
+    {
+      // Declare some data structures
+      std::vector<Point<dim>> quadrature_point_locations;
+      auto                    exadg_terminal_ale =
+        std::make_shared<ExaDG::preCICE::InterfaceCoupling<dim, dim, Number>>();
+
+      // Poisson mesh movement
+      if(this->application->get_parameters_fluid().mesh_movement_type ==
+         IncNS::MeshMovementType::Poisson)
+      {
+        std::vector<unsigned int> quad_indices;
+        if(this->application->get_parameters_ale_poisson().spatial_discretization ==
+           Poisson::SpatialDiscretization::DG)
+          quad_indices.emplace_back(ale_poisson_operator->get_quad_index());
+        else if(this->application->get_parameters_ale_poisson().spatial_discretization ==
+                Poisson::SpatialDiscretization::CG)
+          quad_indices.emplace_back(ale_poisson_operator->get_quad_index_gauss_lobatto());
+        else
+          AssertThrow(false, ExcNotImplemented());
+
+        quadrature_point_locations = exadg_terminal_ale->setup(
+          ale_matrix_free,
+          ale_poisson_operator->get_dof_index(),
+          quad_indices,
+          this->application->get_boundary_descriptor_ale_poisson()->dirichlet_cached_bc);
+      }
+      // Elasticity mesh movement
+      else if(this->application->get_parameters_fluid().mesh_movement_type ==
+              IncNS::MeshMovementType::Elasticity)
+      {
+        std::vector<unsigned int> quad_indices;
+        quad_indices.emplace_back(ale_elasticity_operator->get_quad_index_gauss_lobatto());
+
+        quadrature_point_locations = exadg_terminal_ale->setup(
+          ale_matrix_free,
+          ale_elasticity_operator->get_dof_index(),
+          quad_indices,
+          this->application->get_boundary_descriptor_ale_elasticity()->dirichlet_cached_bc);
+      }
+      else
+      {
+        AssertThrow(false, ExcNotImplemented());
+      }
```

Comment:
With PR #162, this should no longer be necessary: ask spatial operator `get_container_interface_data()`, where `ContainerInterfaceData` can be asked for the array of quadrature points in a second step.

Reply:
@DavidSCN, regarding your comment https://github.com/exadg/exadg/pull/148#issuecomment-1056615634: Could the problem be solved in a way that the spatial operator takes over the responsibility (similar to #162), e.g. the spatial operator fills certain data structures and passes them to precice?

Reply:
We could also set up the dof - ID map in the operator, but it seems more intrusive to me than the current solution.

Reply:
I created issue #204.